CommonTools/RecoAlgos/interface/RecoTrackSelector.h

```diff
@@ -25,10 +26,12 @@ class RecoTrackSelector: public RecoTrackSelectorBase {
     init(event,es);
     selected_.clear();
     for( reco::TrackCollection::const_iterator trk = c->begin();
-         trk != c->end(); ++ trk )
-      if ( operator()(*trk) ) {
+         trk != c->end(); ++ trk ) {
+      reference_type tkref(c,std::distance(c->begin(),trk));
```

Comment:
what's the cost of this cycle to make a ref and then dereference it (in order to pass the same collection pointer to the method downstream)?